Mastodon/Scene/Settings/Privacy and Safety/PrivacySafetyViewModel.swift

```diff
@@ -0,0 +1,225 @@
+        case .custom:
+            break
+        }
+    }
+    
+    func evaluatePreset() {
+        guard !doNotEvaluate else { return }
+        if PrivacySafetySettingPreset.openPublic.equalsSettings(of: self) {
+            preset = .openPublic
+        } else if PrivacySafetySettingPreset.privateRestricted.equalsSettings(of: self) {
+            preset = .privateRestricted
+        } else {
+            preset = .custom
+        }
+    }
+
+    private func loadSettings() {
+        Task { @MainActor in
+            guard let appContext, let authContext else {
+                return dismiss()
+            }
+            
+            let domain = authContext.mastodonAuthenticationBox.domain
+            let userAuthorization = authContext.mastodonAuthenticationBox.userAuthorization
+            
+            let account = try await appContext.apiService.accountVerifyCredentials(
+                domain: domain,
+                authorization: userAuthorization
+            ).singleOutput().value
+            
+            if let privacy = account.source?.privacy {
+                visibility = .from(privacy)
+            }
+            
+            manuallyApproveFollowRequests = account.locked == true
+            showFollowersAndFollowing = account.source?.hideCollections == false
+            suggestMyAccountToOthers = account.source?.discoverable == true
+            appearInSearches = account.source?.indexable == true
+
+            isUserInteractionEnabled = true
+        }
+    }
+    
+    func saveSettings() {
+        Task {
+            guard let appContext, let authContext else {
+                return
+            }
+    
+            let domain = authContext.mastodonAuthenticationBox.domain
+            let userAuthorization = authContext.mastodonAuthenticationBox.userAuthorization
+            
+            let _ = try await appContext.apiService.accountUpdateCredentials(
+                domain: domain,
+                query: .init(
+                    discoverable: suggestMyAccountToOthers,
+                    locked: manuallyApproveFollowRequests,
+                    source: .withPrivacy(visibility.toPrivacy()),
+                    indexable: appearInSearches,
+                    hideCollections: !showFollowersAndFollowing
+                ),
+                authorization: userAuthorization
+            )
+        }
+    }
+    
+    func dismiss() {
+        onDismiss.send(())
+    }
+}
+
+// Preset Rules Definition
+extension PrivacySafetyViewModel {
+    static let openPublic: PrivacySafetySettingApplicable = {
+        let vm = PrivacySafetyViewModel(appContext: nil, authContext: nil, coordinator: nil)
+        vm.visibility = .public
+        vm.manuallyApproveFollowRequests = false
+        vm.showFollowersAndFollowing = true
+        vm.suggestMyAccountToOthers = true
+        vm.appearInSearches = true
+        return vm
+    }()
+    
+    static let privateRestricted: PrivacySafetyViewModel = {
```

Comment:
I wondered why you used the protocol above, but here the ViewModel?

```suggestion
    static let privateRestricted: PrivacySafetySettingApplicable = {
```

Reply:
That was actually a leftover this whole code, except from the `apply(from:)` method, can go . I'll remove it.